rllib/algorithms/a3c/a3c.py

```diff
@@ -250,33 +235,14 @@ def sample_and_compute_grads(worker: RolloutWorker) -> Dict[str, Any]:
             # Synch updated weights back to the particular worker
             # (only those policies that are trainable).
             with self._timers[SYNCH_WORKER_WEIGHTS_TIMER]:
-                weights = local_worker.get_weights(
-                    policies=local_worker.get_policies_to_train()
-                )
-                worker.set_weights.remote(weights, global_vars)
+                self.workers.sync_weights(to_worker_indices=[actor_id])
```

Comment:
Wait, the policies info is lost here! We need to make sure we can control this from the algorithm (not just from the worker set). The algorithm is the only one that knows, which policies to update. It's often not even the trainable ones (which the worker set could figure out), but even fewer, namely those that just got updated by the algorithm.

Reply:
ah, good catch. ok, I made sure the logics are the same.